content/articles/getting-started-with-ray-tracing-algorithm-in-machine-learning/index.md

```diff
@@ -0,0 +1,72 @@
+Ray tracing has been of great impact in the world today, modifying and innovating various sectors of the tech world. In this article, we'll get to know about ray tracing algorithm in machine learning.
+
+### Table of contents
+- [What is Ray Tracing Algorithm](#what-is-ray-tracing-algorithm)
+- [History of Ray Tracing](#history-of-ray-tracing)
+- [How Ray Tracing works](how-ray-tracing-works)
+- [Types of Ray Tracing Algorithm](#types-of-ray-tracing-algorithm)
+  - [Hybrid Ray Tracing Algorithm ](#hybrid-ray-tracing-algorithm)
+  - [Forward Ray Tracing Algorithm ](#forward-ray-tracing-algorithm)
+  - [Backward Ray Tracing Algorithm](#backward-ray-tracing-algorithm) 
+ - [Conclusion](#conclusion)
+ - [Resources](#resources)
+
+### What is Ray Tracing Algorithm 
+Ray Tracing Algorithm is an effective way of drawing synthesizing images. It is a method of rendering which has the potential to produce a much greater degree of authenticity that is probably higher than that of other scanners. However, this involves a huge calculation cost. 
+
+### History of Ray Tracing
+It was created by Robert Goldstein and Arthur Appel in the 1960s. According to [Wikipedia](https://en.wikipedia.org/wiki/Ray_tracing_(graphics)#Algorithm_overview), in 1968, Arthur Appel was the first to use a computer to generate shaded images using ray tracing. He used Rayr Tracing to figure out which surface was closest to the camera at each image point, then traced secondary rays to the source of light from each shaded point to see if it was in shadow or not.
+
+Robert Goldstein then used ray tracing is to create shaded images of solids by simulating the photographic process in reverse. He used a non-recursive ray tracing based algorithm known as ***ray casting***. Ray tracing is widely regarded as the most effective method of image synthesis.
+
+### How Ray Tracing works
+Computer imagery has become a very basic need in today's digital world. Computer graphics come into play when we need to create an image that looks real and indistinguishable from the standard image taken. This can be accomplished in a variety of ways. Ray Tracing is one such way to give. This is used to produce an image by tracing the light path such as pixels in an image plane and mimicking its effects when encountering objects.
+
+The process aims to bring real light to the simulation by imitating the behavior of light intensity. It is done by reflecting the light rays from the camera, ‘following’ them until they encounter an object, and counting where each ray of light cuts an object in the square. This then allows it to calculate how the pixel is affected by the light rays, and to keep track of the ray as it jumps on the scene. This requires the computer to track a certain number of rays from each intersection to the reflected light model, thus creating a growing number of rays in the square.
+
+ ![Ray tracing](/engineering-education/getting-started-with-ray-tracing-algorithm-in-machine-learning/tracing.png)

```

Comment:
Use the diagram to illustrate how ray tracing works. Give explanation while referring to the diagram.